File: src/main/java/net/gamershub/kitpvp/items/impl/EnderSword.java
package net.gamershub.kitpvp.items.impl;

import net.gamershub.kitpvp.abilities.Ability;
import net.gamershub.kitpvp.abilities.AbilityHandler;
import net.gamershub.kitpvp.items.CustomItem;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.format.TextDecoration;
import org.bukkit.Material;
import org.bukkit.enchantments.Enchantment;
import org.bukkit.inventory.ItemStack;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public class EnderSword extends CustomItem {
    public EnderSword() {
        super(Material.STONE_SWORD, "ender_swword", false);
    }

    @Override
    public @NotNull Component getName() {
        return Component.text("Ender Sword").decoration(TextDecoration.ITALIC, false);
    }

    @Override
    public @NotNull List<Ability> getAbilities() {
        return List.of(AbilityHandler.ENDER_ATTACK);
    }

    @Override
    protected void additionalModifications(ItemStack itemStack) {
        itemStack.addEnchantment(Enchantment.DAMAGE_ALL, 3);
    }
}

File: src/main/kotlin/io/github/chaosdave34/kitpvp/items/impl/EnderSword.kt
package io.github.chaosdave34.kitpvp.items.impl

import io.github.chaosdave34.kitpvp.items.CustomItem
import org.bukkit.Material
import org.bukkit.enchantments.Enchantment
import org.bukkit.inventory.ItemStack

class EnderSword : CustomItem(Material.STONE_SWORD, "ender_sword", "Ender Sword") {
    override fun additionalModifications(itemStack: ItemStack) {
        itemStack.addEnchantment(Enchantment.SHARPNESS, 3)
        itemStack.setCustomModelData(1)
    }
}